From a1ca605ad8a4fd1aa76fff259080963f342e07c7 Mon Sep 17 00:00:00 2001
From: Christophe de Dinechin <cdupontd@redhat.com>
Date: Thu, 9 Feb 2017 13:00:19 +0100
Subject: [PATCH 18/26] Show errors generated by connection dialog

When running 'remote-viewer' without arguments,
and selecting a non-supported protocol, e.g. ssh://foo,
the generated error was silently swallowed by the retry loop.
This was introduced in 06b2c382468876a19600374bd59475e66d488af8.

Signed-off-by: Christophe de Dinechin <cdupontd@redhat.com>
---
 src/remote-viewer.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/remote-viewer.c b/src/remote-viewer.c
index d04dbb5..8c31532 100644
--- a/src/remote-viewer.c
+++ b/src/remote-viewer.c
@@ -1195,6 +1195,9 @@ cleanup:
     type = NULL;
 
     if (!ret && priv->open_recent_dialog) {
+        if (error != NULL) {
+            virt_viewer_app_simple_message_dialog(app, _("Unable to connect: %s"), error->message);
+        }
         g_clear_error(&error);
         goto retry_dialog;
     }
-- 
2.12.0

